File: sdk/src/main/java/tv/remo/android/controller/sdk/utils/ValueUtil.java
package tv.remo.android.controller.sdk.utils;

public class ValueUtil {
    public static float map(float input, float inMin, float inMax, float outMin, float outMax, float multiplier) {
        input = range(input, inMin, inMax);
        input *= multiplier;
        inMin *= multiplier;
        inMax *= multiplier;
        float output = (input - inMin) / (inMax - inMin) * (outMax - outMin) + outMin;
        return range(output, outMin, outMax);
    }

    public static float range(float input, float min, float max) {
        float output = input;
        if (input < min) {
            output = min;
        } else if (input > max) {
            output = max;
        }
        return output;
    }
}

File: sdk/src/main/java/tv/remo/android/controller/sdk/utils/ValueUtil.kt
package tv.remo.android.controller.sdk.utils

import java.util.*

object ValueUtil {
    fun randomFloatRange(range: Pair<Float, Float>, random : Random = Random()) : Float{
        return map(random.nextFloat(), Pair(0f, 1f), range)
    }

    fun map(input: Float, inRange: Pair<Float, Float>, outRange: Pair<Float, Float>): Float {
        val boundedInput = range(input, inRange.first, inRange.second)
        val output = (boundedInput - inRange.first) /
                (inRange.second - inRange.first) *
                (outRange.second - outRange.first) + outRange.first
        return range(output, outRange.first, outRange.second)
    }

    fun range(input: Float, min: Float, max: Float): Float {
        var output = input
        if (input < min) {
            output = min
        } else if (input > max) {
            output = max
        }
        return output
    }
}